internal/tiltfile/docker_compose.go

```diff
@@ -335,20 +335,26 @@ func (s *tiltfileState) renameDCService(projectName, name, newName string, svc *
 		return "", err
 	}
 
-	s.dc[projectName].services[newName] = svc
-	delete(s.dc[projectName].services, name)
-	if opts, exists := s.dc[projectName].resOptions[name]; exists {
-		s.dc[projectName].resOptions[newName] = opts
-		delete(s.dc[projectName].resOptions, name)
+	project := s.dc[projectName]
+	services := project.services
+
+	services[newName] = svc
+	delete(services, name)
+	if opts, exists := project.resOptions[name]; exists {
+		project.resOptions[newName] = opts
+		delete(project.resOptions, name)
 	}
 	index := -1
-	for i, n := range s.dc[projectName].serviceNames {
-		if n == name {
+	for i, n := range project.serviceNames {
+		if n == name && index == -1 {
 			index = i
-			break
+		} else if sd, ok := services[n].ServiceConfig.DependsOn[name]; ok {
+			services[n].ServiceConfig.DependsOn[newName] = sd
+			services[n].Options.resourceDeps = []string{newName}
```

Comment:
hmmm...i don't think this is right if resourceDeps has other elements?

Reply:
Oh, good call. I'll fix that when I add the tests (probably in a day or two)

Reply:
Are resource dependencies ordering nondeterministic somehow? I changed this to 
```suggestion
			if rdIndex := slices.Index(services[n].Options.resourceDeps, name); rdIndex != -1 {
				services[n].Options.resourceDeps[rdIndex] = newName
			}
```

I have created several test case permutations; sometimes they all pass, sometimes one or two fail, and it's even inconsistent on which one fails. I'm not sure why it would be flaky.
`f.assertNextManifest(model.ManifestName("bar2"), resourceDeps("db", "foo"))`

Sometimes the tests pass, sometimes they fail with this:
```
Not equal: 
        	            	expected: []model.ManifestName{"db", "foo"}
        	            	actual  : []model.ManifestName{"foo", "db"}
```

Since I'm replacing at the same index, I don't understand what might be causing the order to change. In fact, this seems to happen when the service itself is renamed, even when the dependencies aren't renamed.

My best guess is that this is indirectly due to there being a map involved somewhere in the docker compose code (before it even gets to my changes) and "the iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next," but there's a lot to look through.

Reply:
can you push the version of the code that's failing?

Reply:
looks like it reproed in your CircleCI on the first time, without needing a re-run.

Reply:
can you try rebasing on https://github.com/tilt-dev/tilt/pull/6335 ?

Reply:
I cherry-picked that commit & ran my tests 5 times in a row w/o clearing cache and they all passed!

If you don't mind a force-push, I can push a rebase later when I have a bit more time, but I'm confident that solved the test flakiness, since I never managed twice in a row before.

Reply:
I went ahead and did that, since it seemed that was probably what you wanted.

Reply:
looks like the test that failed is probably a flaky test - it's k8s related, and I didn't touch anything that wasn't docker specific